algebra-core/src/fields/mod.rs

```diff
@@ -158,10 +160,30 @@ pub trait Field:
     }
 }
 
-/// A trait that defines parameters for a prime field.
-pub trait FpParameters: 'static + Send + Sync + Sized {
+/// A trait that defines parameters for a field that can be used for FFT.
+pub trait FftParameters: 'static + Send + Sync + Sized {
     type BigInt: BigInteger;
 
+    /// 2^s * t = MODULUS - 1 with t odd. This is the two-adicity of
+    /// `Self::MODULUS`.
+    const TWO_ADICITY: u32;
+
+    /// 2^s root of unity computed by GENERATOR^t
+    const ROOT_OF_UNITY: Self::BigInt;
+
+    /// The base of a small subgroup that can be used for mixed-radix FFT.
+    const SMALL_SUBGROUP_BASE: Option<u32> = None;
+
+    /// The power of a small subgroup that can be used for mixed-radix FFT.
+    const SMALL_SUBGROUP_POWER: Option<u32> = None;
+
+    /// GENERATOR^((MODULUS-1) / (2^s *
+    /// SMALL_SUBGROUP_BASE^SMALL_SUBGROUP_POWER)) Used for mixed-radix FFT.
+    const FULL_ROOT_OF_UNITY: Option<Self::BigInt> = None;
```

Comment:
Like above, maybe we should change this to `LARGE_SUBGROUP_ROOT_OF_UNITY`/`LARGE_SUBGROUP_GENERATOR`?